src/abstract_solver.h

```diff
@@ -88,10 +88,15 @@ class abstract_solver {
 
   // ******************************************************************
   // solve the problem (must return a value > 0 if a solution has been found, 0
-  // if unsatisfiable, < 0 on error. -2 means timeout)
+  // if unsatisfiable, < 0 on error. -2 means timeout, -3 means sigint)
   virtual int solve() { return 0; }; 
   virtual int solve(int timeout) { return 0; };
 
+  // ******************************************************************
+  // abort should, if possible, do something which causes solve to
+  // terminate.
+  virtual void abort(void) {return;}
```

Comment:
Possibly this should throw some kind of implementation exception. I can't remember if there's a way in C++ to have class members which must be overridden? The point would be to have a definitely way of noticing when a future solver is reintegrated that `abort` support needs adding.

Reply:
`virtual void abort() = 0;`

Reply:
I make no apology for my lack of C++ skills - feel free to make a PR :smile:

(More seriously, I think this is a stylistic correction only, or does it actually cause a problem?)

Reply:
Oh, hang on, the context was really unclear from the GitHub email - you mean that's the answer to my question of how to have a method which must be overridden? In which case, thanks!